src/providers/hover.ts

```diff
@@ -0,0 +1,100 @@
+import * as vscode from "vscode";
+import * as fs from "fs/promises";
+import * as path from "path";
+
+interface HoverData {
+  [keyword: string]: {
+    [block: string]: {
+      [description: string]: string; // dfn_name
+    };
+  };
+}
+
+export class MF6HoverProvider implements vscode.HoverProvider {
+  private hoverData: HoverData = {};
+
+  constructor() {
+    this.loadHoverData();
+  }
+
+  private async loadHoverData() {
+    const filePath = path.resolve(__dirname, "../../src/providers/hover.json");
+    const fileContent = await fs.readFile(filePath, "utf-8");
+    this.hoverData = JSON.parse(fileContent);
+  }
+
+  public async provideHover(
+    document: vscode.TextDocument,
+    position: vscode.Position,
+  ) {
+    const wordRange = document.getWordRangeAtPosition(position, /\S+/);
+    if (!wordRange) {
```

Comment:
```suggestion
    if (!this.hoverData || !wordRange) {
```

Since `loadHoverData` is async, you are not ensured the presence of `this.hoverData` when `provideHover` is called.

Reply:
With https://github.com/martclanor/vscode-mf6-syntax/pull/66/commits/220d4c7d7d2ae19ba6f5c361384e5e84fb29f690, the async `loadHoverData` is removed.